core/templates/rid_owner.h

```diff
@@ -294,43 +287,48 @@ class RID_Alloc : public RID_AllocBase {
 	}
 	void get_owned_list(List<RID> *p_owned) const {
 		if (THREAD_SAFE) {
-			spin_lock.lock();
+			mutex.lock();
 		}
 		for (size_t i = 0; i < max_alloc; i++) {
-			uint64_t validator = validator_chunks[i / elements_in_chunk][i % elements_in_chunk];
+			uint64_t validator = chunks[i / elements_in_chunk][i % elements_in_chunk].validator;
 			if (validator != 0xFFFFFFFF) {
 				p_owned->push_back(_make_from_id((validator << 32) | i));
 			}
 		}
 		if (THREAD_SAFE) {
-			spin_lock.unlock();
+			mutex.unlock();
 		}
 	}
 
 	//used for fast iteration in the elements or RIDs
 	void fill_owned_buffer(RID *p_rid_buffer) const {
 		if (THREAD_SAFE) {
-			spin_lock.lock();
+			mutex.lock();
 		}
 		uint32_t idx = 0;
 		for (size_t i = 0; i < max_alloc; i++) {
-			uint64_t validator = validator_chunks[i / elements_in_chunk][i % elements_in_chunk];
+			uint64_t validator = chunks[i / elements_in_chunk][i % elements_in_chunk].validator;
 			if (validator != 0xFFFFFFFF) {
 				p_rid_buffer[idx] = _make_from_id((validator << 32) | i);
 				idx++;
 			}
 		}
 		if (THREAD_SAFE) {
-			spin_lock.unlock();
+			mutex.unlock();
 		}
 	}
 
 	void set_description(const char *p_descrption) {
 		description = p_descrption;
 	}
 
-	RID_Alloc(uint32_t p_target_chunk_byte_size = 65536) {
+	RID_Alloc(uint32_t p_target_chunk_byte_size = 65536, uint32_t p_maximum_amount_of_elements = 200000) {
```

Comment:
Out of curiosity, shouldn't the default value for `p_maximum_amount_of_elements` be a multiple of `65536`?

Reply:
yeah thats a good point